### PR TITLE
Sitemap functionality addition

### PR DESCRIPTION
**Pull request by sabdulm**
Added sitemap functionality using the next-sitemap package.

### CODE REVIEW

**Comment by sabdulm** [comment]
Add an environment variable on Vercel by the name of `SITE_URL` with the value `https://razakhawaja.me/`

File: next-sitemap.config.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/** @type {import('next-sitemap').IConfig} */
+// Default code you can customize according to your requirements.
+module.exports = {
+    siteUrl: process.env.SITE_URL || 'https://razakhawaja.me',
```
Comment (hamza1220) [inline next-sitemap.config.js:4]: Can we pick this from a public json?

File: next-sitemap.config.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/** @type {import('next-sitemap').IConfig} */
+// Default code you can customize according to your requirements.
```
Comment (hamza1220) [inline next-sitemap.config.js:2]: Lets remove the comments

**Review by hamza1220 — CHANGES_REQUESTED** [review]

**Comment by hamza1220** [comment]
Also, can you convert this into a .ts file?

**Comment by sabdulm** [comment]
> Also, can you convert this into a .ts file?

No, currently next-sitemap is only able to .js configuration files.

File: next-sitemap.config.js (hunk 1 above)
Comment (sabdulm) [reply next-sitemap.config.js:4]: The easiest and recommended way to do this is to have a .env.local file from where we pick the site url and add the env variable to vercel. I can change the hardcoded url to localhost or simply remove it.

Comment (hamza1220) [reply next-sitemap.config.js:4]: Glanced over the env variable and missed it. Changing the hardcoded url to localhost sounds great

**Comment by sabdulm** [comment]
> Add an environment variable on Vercel by the name of `SITE_URL` with the value `https://razakhawaja.me/`

Before merging, add the ENV variable and redeploy it on Preview to view the correct sitemap at https://razakhawaja.me/sitemap.xml

**Review by hamza1220 — APPROVED** [review]

**Pull request merged**